worker/worker.go

```diff
@@ -1353,36 +1360,48 @@ func (w *worker) Handler() http.Handler {
 		"PUT    /multipart/*path": w.multipartUploadHandlerPUT,
 
 		"GET    /state": w.stateHandlerGET,
-	}))
+	})))
 }
 
 // Shutdown shuts down the worker.
-func (w *worker) Shutdown(_ context.Context) error {
-	w.interactionsMu.Lock()
-	if w.interactionsFlushTimer != nil {
-		w.interactionsFlushTimer.Stop()
-		w.flushInteractions()
-	}
-	w.interactionsMu.Unlock()
+func (w *worker) Shutdown(ctx context.Context) error {
+	// cancel shutdown context
+	w.shutdownCtxCancel(ErrShuttingDown)
 
-	// Cancel shutdown context.
-	w.shutdownCtxCancel()
+	// stop recorders
+	w.hostInteractionRecorder.Stop(ctx)
+	w.contractSpendingRecorder.Stop(ctx)
```

Comment:
Shouldn't those be closed after the upload and download managers considering that those might cause interactions and spendings?